src/shapes/text.class.js

```diff
@@ -365,14 +365,14 @@
      * @return {fabric.Text} thisArg
      */
     initialize: function(text, options) {
-      this.styles = options ? (options.styles || { }) : { };
       this.text = text;
       this.__skipDimension = true;
       this.callSuper('initialize', options);
       if (this.path) {
         this.setPathInfo();
       }
       this.__skipDimension = false;
+      this.styles = fabric.util.stylesFromArray(options ? (options.styles || { }) : { }, this.text);
```

Comment:
i think this conversion should go in the `fromObject` function more than initialize.
Reason: if the runtime style format for now stays the same, we want to still be able to initialize a new text copying the style from another.
If we go though the clone function nothing will change since we use toObject and fromObject.
In general since this is an object dump change, i would like to handle into the toObject/fromObject pair

Reply:
done.